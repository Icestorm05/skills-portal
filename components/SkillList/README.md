# Skill List

## Functionality

The skill list component displays a recursive list of all the skill categories and skills. Each skill category is displayed in an expansion panel. Clicking on this panel will display the skills and/or skill categories within the category. Content inside of each expansion panel is lazily rendered, in order to optimize performance.

## Usage

### Props

`<skill-list>` supports the following custom component attributes:

| attribute | type | description
| --- | --- | ---
| `lazy` | Boolean *optional* | Whether to render the skill list lazily or not. Defaults to `true`.
| `skills` | Array | The array of skill and skill category objects.

### Slots

`<skill-list>` exposes the following slots:

| slot | scope | type | description
| --- | --- | ---
| `default` | `skill` | Object | The skill object at the bottom level of the array passed in. Can either be a skill, saved skill or approved skill schema object.

### Schemas

The `skills` prop requires either of the following objects to exist within the array:

#### Skill Category Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill category.
| `children` | Array | An array of further skill or skill categories. **Must** eventually lead down to a skill or saved skill schema object.

#### Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The unique attribute ID of the skill.

#### Saved Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The attribute ID of the skill.
| `EmployeeSkillAttributeId` | Number | A link to the SkillAttributeId.
| `EmployeeId` | Number | The ID of the employee who has the skill.
| `EmployeeSkillId` | Number | The unique ID of the saved skill.
| `SkillCurrencyId` | Number | The ID of the currency. 2 maps to `Current`. 3 maps to `Fading`. 4 maps to `Historic`.
| `SkillLevelId` | Number | The ID of the level. 2 maps to `Awareness`. 3 maps to `Practitioner`. 4 maps to `Expert`.

#### Approved Skill Schema

| attribute | type | description
| --- | --- | ---
| `name` | String | The name of the skill.
| `description` | String *optional* | The description of the skill.
| `SkillAttributeId` | Number | The attribute ID of the skill.
| `EmployeeApproveSkillAttributeId` | Number | A link to the SkillAttributeId.
| `EmployeeApprovedSkillId` | Number | The unique ID of the approved skill.
| `EmployeeId` | Number | The ID of the employee who has the skill.
| `SkillCurrencyId` | Number | The ID of the currency. 2 maps to `Current`. 3 maps to `Fading`. 4 maps to `Historic`.
| `SkillLevelId` | Number | The ID of the level. 2 maps to `Awareness`. 3 maps to `Practitioner`. 4 maps to `Expert`.

### Example

```html
<skill-list :skills="skills">
    <p slot-scope="{ skill }">{{ skill.name }}</p>
</skill-list>
```

```ts
import Component from 'nuxt-class-component';
import Vue from 'vue';

import SkillList from '~/components/SkillList/SkillList.vue';

@Component({
    components: {
        SkillList,
    },
})
export default class GenericComponent extends Vue {
    private skills = [{
        name: 'Test Category',
        children: [{
            name: 'Test Category 2',
            children: [{
                name: 'Test Skill',
                description: 'This is a test skill.',
                SkillAttributeId: 1,
            }, {
                name: 'Test Skill 2',
                description: 'This is another test skill.',
                SkillAttributeId: 2,
            }],
        }],
    }]
}
```