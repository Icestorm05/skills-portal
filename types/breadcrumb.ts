import { Location } from 'vue-router';

export interface IBreadcrumb {
    link: Location;
    text: string;
}