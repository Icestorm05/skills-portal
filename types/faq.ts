export interface IFaq {
    question: string;
    answers: IFaqAnswer[];
}

export interface IFaqAnswer {
    type: 'p' | 'b' | 'ul' | 'table';
    text: string | string[] | string[][];
}