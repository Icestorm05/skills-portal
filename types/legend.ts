export interface ILegend {
    SkillCategories: ISkillCategoryLegend;
    Levels: ISkillLevelLegend;
    Currencies: ISkillCurrencyLegend;
    fallback: string;
}

export interface ISkillCategoryLegend {
    [key: string]: string;
}

export interface ISkillLevelLegend {
    Awareness: string;
    Practitioner: string;
    Expert: string;
}

export interface ISkillCurrencyLegend {
    Historic: string;
    Fading: string;
    Current: string;
}