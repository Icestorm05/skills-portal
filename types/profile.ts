export interface IProfile {
    Biography: string;
    Company: string;
    Director: string;
    EmailAddress: string;
    EmployeeId: number;
    FirstLineReporting: string;
    FullName: string;
    JobTitle: string;
    KnownAs: string;
    MainStaffNumber: string;
    SecondLineReporting: string;
    Surname: string;
    children: IProfile[];
}