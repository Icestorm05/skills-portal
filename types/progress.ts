export interface IProgress {
    display: boolean;
}