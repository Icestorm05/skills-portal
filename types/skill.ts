import { ECurrency } from 'enums/currency';
import { ELevel } from 'enums/level';

export interface IDescription {
    name: string;
    description: string;
}

export interface ISkillCategory extends IDescription {
    AreaId?: number;
    SubAreaId?: number;
    children: (ISkill|ISkillCategory)[];
}

export interface ISkill extends IDescription {
    SkillAttributeId: number;
    Area: string;

    EmployeeSkillId?: number;
    EmployeeId?: number;
    EmployeeSkillAttributeId?: number;
    SkillCurrencyId?: number;
    SkillLevelId?: number;

    EmployeeApprovedSkillId?: number;
    EmployeeApprovedSkillAttributeId?: number;

    path?: string[];
}
