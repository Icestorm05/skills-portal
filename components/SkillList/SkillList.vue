<template lang="pug">
    v-container(fluid
                grid-list-md).SkillList.pt-0.px-0
        v-expansion-panel(v-if="skillCategoriesOnly.length"
                          expand
                          popout).SkillPanels
            v-expansion-panel-content(v-for="category in skillCategoriesOnly"
                                      :key="category.name"
                                      :lazy="lazy"
                                      ripple).SkillPanel.mb-2
                span(slot="header").white--text.SkillPanelHeader {{ category.name }}
                p.ma-3 {{ category.description }}
                skill-list(:skills="category.children"
                           :lazy="lazy").pl-3
                    template(slot-scope="{ skill }")
                        slot(:skill="skill")
        v-data-iterator(v-if="skillsOnly.length"
                        content-class="layout row wrap fill-width"
                        :rows-per-page-items="[3, 6, 12, 24, {'text': 'All', 'value': -1}]"
                        rows-per-page-text="Skills per page:"
                        :items="skillsOnly"
                        :pagination="pagination"
                        @update:pagination="updatePagination"
                        ref="dataIterator").Skills
            v-flex(slot="item"
                    slot-scope="{ item }"
                    sm4).SkillContainer
                slot(:skill="item")
</template>

<script lang="ts">
import Component from "nuxt-class-component";
import Vue from "vue";
import { Prop, Watch } from "vue-property-decorator";

import { ISkillCategory } from "types/skill";

import VDataIterator from "vuetify/es5/components/VDataIterator";
import VExpansionPanel, {
  VExpansionPanelContent
} from "vuetify/es5/components/VExpansionPanel";
import { VContainer, VFlex, VLayout } from "vuetify/es5/components/VGrid";

@Component({
  components: {
    VContainer,
    VDataIterator,
    VExpansionPanel,
    VExpansionPanelContent,
    VFlex,
    VLayout
  },
  name: "skill-list"
})
export default class SkillList extends Vue {
  @Prop({ type: Boolean, default: true })
  private lazy: boolean;
  @Prop({ type: Array, required: true })
  private skills: ISkillCategory[];

  private get skillCategoriesOnly() {
    return this.skills.filter(skill => (skill.children ? true : false));
  }

  private pagination = {
    rowsPerPage: 3,
    page: 1,
    totalItems: this.skillsOnly.length
  };

  private overridePaginationUpdate: boolean = false;
  private updatePagination(pagination) {
    if (!this.overridePaginationUpdate) {
      this.pagination = pagination;
    } else {
      const page = this.pagination.page;
      this.pagination = pagination;
      Vue.set(this.pagination, "page", page);
      this.overridePaginationUpdate = false;
    }
  }

  private get skillsOnly() {
    return this.skills.filter(skill => (skill.children ? false : true));
  }

  @Watch("skillsOnly")
  private skillsOnlyChanged(curr, prev) {
    if (prev.length !== curr.length) {
      this.overridePaginationUpdate = true;
      const lastPage = Math.ceil(curr.length / this.pagination.rowsPerPage);
      this.pagination.page =
        (this.pagination.page > lastPage
          ? this.pagination.page - 1
          : this.pagination.page) || 1;
    }
  }
}
</script>

<style lang="stylus" scoped>
theme = json('../../assets/json/theme.json', { hash: true });

.SkillPanel {
  &.expansion-panel__container--active {
    margin-top: 8px;
    margin-left: 0;
    margin-right: 0;
  }

  background-color: lighten(theme.secondary, 57) !important;

  /deep/ .expansion-panel__header {
    background-color: theme.secondary;

    .icon {
      color: rgba(255, 255, 255, 0.54) !important;
    }
  }
}
</style>
