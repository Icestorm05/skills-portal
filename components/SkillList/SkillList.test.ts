import SkillList from "./SkillList.vue";
import { mount } from "@vue/test-utils";

import { ISkill, ISkillCategory } from "types/skill";

declare const expect: jest.Expect;

// Add a wrapping <div data-app="true"></div> to silence Vuetify warnings.
const app = document.createElement("div");
app.setAttribute("data-app", "true");
document.body.appendChild(app);

const skillCategory: ISkillCategory = {
  name: "Test Skill Category 1",
  children: [],
  description: ""
};

const skillCategory2: ISkillCategory = {
  name: "Test Skill Category 2",
  children: [],
  description: ""
};

const skillSubcategory: ISkillCategory = {
  name: "Test Skill Subcategory 1",
  children: [],
  description: ""
};

const skillSubcategory2: ISkillCategory = {
  name: "Test Skill Subcategory 2",
  children: [],
  description: ""
};

const skill: ISkill = {
  SkillAttributeId: 1,
  name: "Test Skill",
  Area: "",
  description: ""
};

const skill2: ISkill = {
  SkillAttributeId: 2,
  name: "Test Skill 2",
  Area: "",
  description: ""
};

const skill3: ISkill = {
  SkillAttributeId: 3,
  name: "Test Skill 3",
  Area: "",
  description: ""
};

const skill4: ISkill = {
  SkillAttributeId: 4,
  name: "Test Skill 4",
  Area: "",
  description: ""
};

const skills: ISkillCategory[] = [
  {
    ...skillCategory,
    children: [
      {
        ...skillSubcategory,
        children: [skill]
      },
      {
        ...skillSubcategory2,
        children: [skill2]
      },
      skill3
    ]
  },
  {
    ...skillCategory2,
    children: [skill4]
  }
];

const skillsFlatMix: Array<ISkillCategory | ISkill> = [
  skillCategory,
  skillCategory2,
  skill,
  skill2
];

describe("Skill List", () => {
  test("is a Vue instance", () => {
    const wrapper = mount(SkillList, { propsData: { skills } });
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("snapshot has not changed", async () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("lazy snapshot has not changed", async () => {
    const wrapper = mount(SkillList, { propsData: { lazy: true, skills } });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("lazy prop set to true only renders first level", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: true, skills } });
    expect(wrapper.find(".SkillPanel").exists()).toBeTruthy();
    expect(wrapper.find(".SkillPanel .SkillPanel").exists()).toBeFalsy();
  });

  test("lazy prop set to false renders all levels", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    expect(wrapper.find(".SkillPanel").exists()).toBeTruthy();
    expect(wrapper.find(".SkillPanel .SkillPanel").exists()).toBeTruthy();
  });

  test("renders the first level categories correctly", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    const headers = wrapper.findAll(".SkillPanel .SkillPanelHeader");
    expect(headers.length).toEqual(4);
    expect(headers.at(0).text()).toEqual(skillCategory.name);
    expect(headers.at(3).text()).toEqual(skillCategory2.name);
  });

  test("renders the second level subcategories correctly", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    const headers = wrapper.findAll(
      ".SkillPanel .SkillPanel .SkillPanelHeader"
    );
    expect(headers.length).toEqual(2);
    expect(headers.at(0).text()).toEqual(skillSubcategory.name);
    expect(headers.at(1).text()).toEqual(skillSubcategory2.name);
  });

  test("renders the second level skills correctly", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    const skillWrappers = wrapper.findAll(".SkillPanel .Skills");
    expect(skillWrappers.length).toEqual(4);
  });

  test("renders the third level correctly", () => {
    const wrapper = mount(SkillList, { propsData: { lazy: false, skills } });
    const skillWrappers = wrapper.findAll(".SkillPanel .SkillPanel .Skills");
    expect(skillWrappers.length).toEqual(2);
  });

  test("renders flat level of just skill categories mix correctly", () => {
    const wrapper = mount(SkillList, {
      propsData: { lazy: false, skills: skillsFlatMix }
    });
    const panelHeaders = wrapper.findAll(".SkillPanel .SkillPanelHeader");
    expect(panelHeaders.length).toEqual(2);
    expect(panelHeaders.at(0).text()).toEqual(skillCategory.name);
    expect(panelHeaders.at(1).text()).toEqual(skillCategory2.name);
  });

  test("joins skills on same level together", () => {
    const wrapper = mount(SkillList, {
      propsData: { lazy: false, skills: skillsFlatMix }
    });
    const skillWrappers = wrapper.findAll(".Skills");
    expect(skillWrappers.length).toEqual(1);
  });

  test("creates a computed property of skill categories only", () => {
    const wrapper = mount(SkillList, {
      propsData: { lazy: false, skills: skillsFlatMix }
    });
    const categoriesOnly = (wrapper.vm as any).skillCategoriesOnly;
    expect(categoriesOnly.length).toEqual(2);
    expect(categoriesOnly).toContain(skillCategory);
    expect(categoriesOnly).toContain(skillCategory2);
  });

  test("creates a computed property of skills only", () => {
    const wrapper = mount(SkillList, {
      propsData: { lazy: false, skills: skillsFlatMix }
    });
    const skillsOnly = (wrapper.vm as any).skillsOnly;
    expect(skillsOnly.length).toEqual(2);
    expect(skillsOnly).toContain(skill);
    expect(skillsOnly).toContain(skill2);
  });

  // TO DO: Test scoped slot
});
