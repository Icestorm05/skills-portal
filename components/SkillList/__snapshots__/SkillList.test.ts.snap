// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill List lazy snapshot has not changed 1`] = `
<div class="container SkillList pt-0 px-0 fluid grid-list-md">
    <ul class="expansion-panel SkillPanels expansion-panel--popout">
        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Category 1</span>
                <div class="header__icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="expansion-panel__body" style="display: none;"></div>
        </li>
        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Category 2</span>
                <div class="header__icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="expansion-panel__body" style="display: none;"></div>
        </li>
    </ul>
    <!---->
</div>
`;

exports[`Skill List snapshot has not changed 1`] = `
<div class="container SkillList pt-0 px-0 fluid grid-list-md">
    <ul class="expansion-panel SkillPanels expansion-panel--popout">
        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Category 1</span>
                <div class="header__icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="expansion-panel__body" style="display: none;">
                <p class="ma-3"></p>
                <div class="container SkillList pt-0 px-0 fluid grid-list-md pl-3">
                    <ul class="expansion-panel SkillPanels expansion-panel--popout">
                        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
                            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Subcategory 1</span>
                                <div class="header__icon">
                                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                                </div>
                            </div>
                            <div class="expansion-panel__body" style="display: none;">
                                <p class="ma-3"></p>
                                <div class="container SkillList pt-0 px-0 fluid grid-list-md pl-3">
                                    <!---->
                                    <div class="Skills data-iterator">
                                        <div class="layout row wrap fill-width">
                                            <div class="flex SkillContainer sm4"></div>
                                        </div>
                                        <div>
                                            <div class="data-iterator__actions">
                                                <div class="data-iterator__actions__select">Skills per page:
                                                    <div tabindex="0" data-uid="16" aria-label="Skills per page:" role="combobox" class="input-group input-group--append-icon input-group--hide-details input-group--text-field input-group--select input-group--auto primary--text">
                                                        <div class="input-group__input">
                                                            <div class="input-group__selections" style="overflow: hidden;">
                                                                <input aria-label="Skills per page:" disabled="disabled" tabindex="-1" class="input-group--select__autocomplete" style="display: none;">
                                                            </div>
                                                            <div class="menu" style="display: inline-block;"></div>
                                                            <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb">arrow_drop_down</i>
                                                        </div>
                                                        <div class="input-group__details"></div>
                                                    </div>
                                                </div>
                                                <div class="data-iterator__actions__range-controls">
                                                    <div class="data-iterator__actions__pagination">1-1 of 1</div>
                                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Previous page">
                                                        <div class="btn__content">
                                                            <i aria-hidden="true" class="icon material-icons">chevron_left</i>
                                                        </div>
                                                    </button>
                                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Next page">
                                                        <div class="btn__content">
                                                            <i aria-hidden="true" class="icon material-icons">chevron_right</i>
                                                        </div>
                                                    </button>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </li>
                        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
                            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Subcategory 2</span>
                                <div class="header__icon">
                                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                                </div>
                            </div>
                            <div class="expansion-panel__body" style="display: none;">
                                <p class="ma-3"></p>
                                <div class="container SkillList pt-0 px-0 fluid grid-list-md pl-3">
                                    <!---->
                                    <div class="Skills data-iterator">
                                        <div class="layout row wrap fill-width">
                                            <div class="flex SkillContainer sm4"></div>
                                        </div>
                                        <div>
                                            <div class="data-iterator__actions">
                                                <div class="data-iterator__actions__select">Skills per page:
                                                    <div tabindex="0" data-uid="32" aria-label="Skills per page:" role="combobox" class="input-group input-group--append-icon input-group--hide-details input-group--text-field input-group--select input-group--auto primary--text">
                                                        <div class="input-group__input">
                                                            <div class="input-group__selections" style="overflow: hidden;">
                                                                <input aria-label="Skills per page:" disabled="disabled" tabindex="-1" class="input-group--select__autocomplete" style="display: none;">
                                                            </div>
                                                            <div class="menu" style="display: inline-block;"></div>
                                                            <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb">arrow_drop_down</i>
                                                        </div>
                                                        <div class="input-group__details"></div>
                                                    </div>
                                                </div>
                                                <div class="data-iterator__actions__range-controls">
                                                    <div class="data-iterator__actions__pagination">1-1 of 1</div>
                                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Previous page">
                                                        <div class="btn__content">
                                                            <i aria-hidden="true" class="icon material-icons">chevron_left</i>
                                                        </div>
                                                    </button>
                                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Next page">
                                                        <div class="btn__content">
                                                            <i aria-hidden="true" class="icon material-icons">chevron_right</i>
                                                        </div>
                                                    </button>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </li>
                    </ul>
                    <div class="Skills data-iterator">
                        <div class="layout row wrap fill-width">
                            <div class="flex SkillContainer sm4"></div>
                        </div>
                        <div>
                            <div class="data-iterator__actions">
                                <div class="data-iterator__actions__select">Skills per page:
                                    <div tabindex="0" data-uid="45" aria-label="Skills per page:" role="combobox" class="input-group input-group--append-icon input-group--hide-details input-group--text-field input-group--select input-group--auto primary--text">
                                        <div class="input-group__input">
                                            <div class="input-group__selections" style="overflow: hidden;">
                                                <input aria-label="Skills per page:" disabled="disabled" tabindex="-1" class="input-group--select__autocomplete" style="display: none;">
                                            </div>
                                            <div class="menu" style="display: inline-block;"></div>
                                            <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb">arrow_drop_down</i>
                                        </div>
                                        <div class="input-group__details"></div>
                                    </div>
                                </div>
                                <div class="data-iterator__actions__range-controls">
                                    <div class="data-iterator__actions__pagination">1-1 of 1</div>
                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Previous page">
                                        <div class="btn__content">
                                            <i aria-hidden="true" class="icon material-icons">chevron_left</i>
                                        </div>
                                    </button>
                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Next page">
                                        <div class="btn__content">
                                            <i aria-hidden="true" class="icon material-icons">chevron_right</i>
                                        </div>
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </li>
        <li tabindex="0" class="expansion-panel__container SkillPanel mb-2">
            <div class="expansion-panel__header"><span class="white--text SkillPanelHeader">Test Skill Category 2</span>
                <div class="header__icon">
                    <i aria-hidden="true" class="icon material-icons">keyboard_arrow_down</i>
                </div>
            </div>
            <div class="expansion-panel__body" style="display: none;">
                <p class="ma-3"></p>
                <div class="container SkillList pt-0 px-0 fluid grid-list-md pl-3">
                    <!---->
                    <div class="Skills data-iterator">
                        <div class="layout row wrap fill-width">
                            <div class="flex SkillContainer sm4"></div>
                        </div>
                        <div>
                            <div class="data-iterator__actions">
                                <div class="data-iterator__actions__select">Skills per page:
                                    <div tabindex="0" data-uid="61" aria-label="Skills per page:" role="combobox" class="input-group input-group--append-icon input-group--hide-details input-group--text-field input-group--select input-group--auto primary--text">
                                        <div class="input-group__input">
                                            <div class="input-group__selections" style="overflow: hidden;">
                                                <input aria-label="Skills per page:" disabled="disabled" tabindex="-1" class="input-group--select__autocomplete" style="display: none;">
                                            </div>
                                            <div class="menu" style="display: inline-block;"></div>
                                            <i aria-hidden="true" class="icon material-icons input-group__append-icon input-group__icon-cb">arrow_drop_down</i>
                                        </div>
                                        <div class="input-group__details"></div>
                                    </div>
                                </div>
                                <div class="data-iterator__actions__range-controls">
                                    <div class="data-iterator__actions__pagination">1-1 of 1</div>
                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Previous page">
                                        <div class="btn__content">
                                            <i aria-hidden="true" class="icon material-icons">chevron_left</i>
                                        </div>
                                    </button>
                                    <button disabled="disabled" type="button" class="btn btn--disabled btn--flat btn--icon" aria-label="Next page">
                                        <div class="btn__content">
                                            <i aria-hidden="true" class="icon material-icons">chevron_right</i>
                                        </div>
                                    </button>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </li>
    </ul>
    <!---->
</div>
`;
